### ISSUE TITLE
Error: Unsupported architecture on Linux: arm64

### ISSUE THREAD

**Issue by dionjwa** [denoland/denokv#71]
## Problem

Environment:
 - host: macos
 - via [nhost functions locally in docker](https://github.com/nhost/functions) 

On installing the `@deno/kv` package I get this runtime error

```
functions-1  | /opt/project/functions/node_modules/@deno/kv/script/_napi_index.cjs:100
functions-1  |         throw new Error(`Unsupported architecture on Linux: ${arch}`)
functions-1  |               ^
functions-1  |
functions-1  | Error: Unsupported architecture on Linux: arm64
functions-1  |     at Object.<anonymous> (/opt/project/functions/node_modules/@deno/kv/script/_napi_index.cjs:100:15)
functions-1  |     at Module._compile (node:internal/modules/cjs/loader:1256:14)

```

Looking into `/opt/project/functions/node_modules/@deno/kv/script/_napi_index.cjs:100` then yes there is no matching package (linux arm64)

This is the environment you can get in docker in macos.

## Solution

Maybe add an automated build for `linux/arm64`

This architecture is also on e.g. raspberry pi devices.

**Comment by dbushell** [comment]
I was able to build the image on a Raspberry Pi 5:

```shell
git clone https://github.com/denoland/denokv.git
cd denokv
docker buildx build -f Dockerfile --platform linux/arm64 -t rpi5/denokv ./
```

Change `rpi5/denokv` to whatever image name you prefer.

Build took around 8 minutes.

On an M2 MacBook I've been using the `linx/amd64` image with no problem. I get the warning: 

```
denokv The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

but it still runs fine. Using Docker Desktop 4.28.0 on macOS 14.4. I did try building a native image (same command above) took a few minutes, haven't bothered to test it though. 

Raspberry Pi 5:
```
Linux rpi5 6.6.20+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
```

Docker version on Pi 5 ([install guide](https://docs.docker.com/engine/install/debian/)):
```
Client: Docker Engine - Community
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:08:02 2023
 OS/Arch:           linux/amd64
 Context:           default
 ```
 
 Running fine on the Pi so far. I'll report back if any problems arise.
 
 Official arm images would be nice now that Deno has offical arm binaries.

**Comment by dionjwa** [comment]
Yeah, it's the npm module `@deno/kv` that doesn't have the architecture. I am using the `@deno/kv` module from a `node.js` runtime. I don't have the ability to run my own docker image in this particular development setup, unfortunately.

**Comment by dionjwa** [comment]
But thanks for that example to create my own docker image.

**Comment by fabricionaweb** [comment]
That is unfortunately, what is needed to have it?

**Comment by johnspurlock-skymethod** [comment]
It's been a while since I looked at this, but it would look something like restoring only the `aarch64-unknown-linux-gnu` parts out of this: https://github.com/skymethod/kv-connect-kit/commit/6ff39d26deca3ab4a1023d4d2523cd293b7e8bb5

and then add the new arch to `scripts/generate_napi_index.ts`

**Comment by dbushell** [comment]
Looks like this was resolve in v8.3 https://github.com/denoland/denokv/commit/ee723057db4ee4c8b24f1adf08340d05b98958e3

The [GitHub releases](https://github.com/denoland/denokv/releases) page is still on 7.0 but the [packages](https://github.com/denoland/denokv/pkgs/container/denokv) are on 8.3 with a `linux/arm64` image.
